### PR TITLE
HomogeneousSampling for chains

### PR DESCRIPTION
**Pull request by ErickChacon**

### CODE REVIEW

File: test/sampling.jl [hunk 1]
```diff
@@ -137,6 +137,12 @@ using Base: need_full_hex
     @test first(ps) isa P2
     @test all(0 ≤ coords[1] == coords[2] ≤ 1 for coords in coordinates.(ps))
 
+    c = Chain(P2(0,0), P2(1,0), P2(0,1), P2(1,1))
+    ps = sample(c, HomogeneousSampling(100))
+    @test first(ps) isa P2
+    @test all(coords[1] + coords[2] == 1 || (0 ≤ coords[1] ≤ 1 && coords[2] ∈ (0, 1))
```
Comment (juliohm) [inline test/sampling.jl]:
```suggestion
    @test all(coords[1] + coords[2] == 1 || (0 ≤ coords[1] ≤ 1 && coords[2] ∈ [0, 1])
```

**Comment by juliohm** [comment]
Awesome! Happy seeing more contributions ❤️  I will just wait for the tests to pass and will merge.

Now we just need a \in method for Segment, willing to submit a PR for that as well? :)

**Comment by ErickChacon** [comment]
> Awesome! Happy seeing more contributions heart I will just wait for the tests to pass and will merge.
> 
> Now we just need a \in method for Segment, willing to submit a PR for that as well? :)

Yes, the basic idea is to use the cross product on R^2, but I am not sure about higher dimensional spaces. Do we probably have to apply the same for each of pair of axis with a total of $n \choose 2$?

**Comment by juliohm** [comment]
Feel free to start with 2D. Maybe our intersection algorithms between two
segments can be useful here too.

Em qui., 28 de out. de 2021 às 11:32, Erick A. Chacón Montalván <
***@***.***> escreveu:

**Comment by ErickChacon** [comment]
Ok, I will check the intersection algorithms you already have.

**Pull request merged**